folium/plugins/semicircle.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import (absolute_import, division, print_function)
+
+import json
+
+from branca.element import Figure, JavascriptLink
+
+from folium.map import Marker
+from folium.utilities import _validate_location
+
+from jinja2 import Template
+
+
+class SemiCircle(Marker):
+    """
+    Creates a Semicircle plugin to add to a map.
+    Use (direction and arc) or (start_angle and stop_angle)
+    Parameters
+    ----------
+    location: tuple of length 2,
+        The latitude and longitude of the marker.
+        If None, then the middle of the map is used.
+    radius: int
+        Radius of semicircle in meters
+    direction: int, default None
+        Heading of direction angle value between 0 and 360 degrees
+    arc: int, default None
+        Heading of arc angle value between 0 and 360 degrees.
+    start_angle: int, default None
+        Heading of the start angle value between 0 and 360 degrees
+    stop_angle: int, default None
+        Heading of the stop angle value between 0 and 360 degrees.
+    fill_color: string, default '#3388ff' (blue)
+        Fill color
+    fill_opacity: float 0-1, default 0.5
+        Fill opacity
+    color: string, default '#3388ff' (blue)
+        Line color
+    opacity: float 0-1, default 1
+        Line opacity
+    **kwargs:
+        Passed to Marker so a popup and tooltip can be provided.
+
+    """
+    _template = Template(u"""
+            {% macro script(this, kwargs) %}
+                if ({{this.direction}} || {{this.arc}}) {
+                    var {{this.get_name()}} = L.semiCircle(
+                        [{{this.location[0]}},{{this.location[1]}}],
+                        {radius:{{this.radius}},
```

Comment:
After using `parse_options` you can put the options in the template like this: `{{ this.options|tojson}}`. Like here: https://github.com/python-visualization/folium/blob/master/folium/plugins/polyline_text_path.py#L49